### PR TITLE
Add integration test for mongodb receiver

### LINKED ISSUE #10864: [receiver/mongodb] Add integration test
The task is essentially to create an integration test that:
1. Starts an mongodb docker container
2. Initializes the mongodbreceiver component from it's factory
3. Scrapes metrics from the docker container
4. Validates that the expected metrics are scraped (using the [scrapertest package](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/scrapertest))

As far as examples go, there are a few components out there that have integration tests using the same pattern.
For instance:

[Mysql receiver integration test](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8eec12f2688309fe4bf0b97ad73128ab12d72861/receiver/mysqlreceiver/integration_test.go)
[Nginx receiver integration test](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8eec12f2688309fe4bf0b97ad73128ab12d72861/receiver/nginxreceiver/integration_test.go)

_Adapted from post by @BinaryFissionGames in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10165#issuecomment-1133257413_

### PR DESCRIPTION
**Pull request by Frapschen**
**Description:** Add integration test for mongodb receiver

Resolves #10864

### CODE REVIEW

File: receiver/mongodbreceiver/ingegration_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+		}, 2*time.Minute, 1*time.Second, "failed to receive more than 0 metrics")
+		require.NoError(t, rcvr.Shutdown(context.Background()))
+
+		actualMtrics := consumer.AllMetrics()[0]
+
+		actualFile := filepath.Join("testdata", "integration", "actual.5_0.json")
+		err = golden.WriteMetrics(actualFile, actualMtrics)
+		require.NoError(t, err)
+
+		expectedFile := filepath.Join("testdata", "integration", "expected.5_0.json")
```
Comment (djaglowski) [inline receiver/mongodbreceiver/ingegration_test.go:132]: How was this file created?

Typically, it should be written from the actual result. (One time during development, and then the `WriteMetrics` call should be removed.) There should not be an "actual metrics" file, only an "expected metrics" file.

File: receiver/mongodbreceiver/ingegration_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		require.Eventuallyf(t, func() bool {
+			return len(consumer.AllMetrics()) > 0
+		}, 2*time.Minute, 1*time.Second, "failed to receive more than 0 metrics")
+		require.NoError(t, rcvr.Shutdown(context.Background()))
+
+		actualMtrics := consumer.AllMetrics()[0]
+
+		actualFile := filepath.Join("testdata", "integration", "actual.5_0.json")
+		err = golden.WriteMetrics(actualFile, actualMtrics)
+		require.NoError(t, err)
```
Comment (djaglowski) [inline receiver/mongodbreceiver/ingegration_test.go:130]: This should be considered temporary code.
```suggestion
		actualFile := filepath.Join("testdata", "integration", "expected.5_0.json")
		err = golden.WriteMetrics(actualFile, actualMtrics)
		require.NoError(t, err)
```

Run this one time and then remove all of these lines. Then you have your "expected" file.

File: receiver/mongodbreceiver/ingegration_test.go [hunk 3]
```diff
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build integration
```
Comment (djaglowski) [inline receiver/mongodbreceiver/ingegration_test.go:15]: Please double check the name of this file.

File: receiver/mongodbreceiver/ingegration_test.go (hunk 2 above)
Comment (Frapschen) [reply receiver/mongodbreceiver/ingegration_test.go:130]: ok, I have known how to do next.

File: receiver/mongodbreceiver/ingegration_test.go (hunk 3 above)
Comment (Frapschen) [reply receiver/mongodbreceiver/ingegration_test.go:15]: ok

**Comment by Frapschen** [comment]
@djaglowski review pls :)

File: receiver/mongodbreceiver/integration_test.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+			Dockerfile: "Dockerfile.mongodb.4_0",
+		},
+		ExposedPorts: []string{"27017:27017"},
+		WaitingFor:   wait.ForListeningPort("27017").WithStartupTimeout(2 * time.Minute),
+	}
+)
+
+func TestMongodbIntegration(t *testing.T) {
+	t.Run("Running mongodb 4.0", func(t *testing.T) {
+		t.Parallel()
```
[the remark this answers is not in the document]
Comment (djaglowski) [reply receiver/mongodbreceiver/integration_test.go:59]:
```suggestion
```

File: receiver/mongodbreceiver/integration_test.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+		expectedFile := filepath.Join("testdata", "integration", "expected.4_0.json")
+		expectedMetrics, err := golden.ReadMetrics(expectedFile)
+		require.NoError(t, err)
+
+		err = scrapertest.CompareMetrics(expectedMetrics, actualMetrics, scrapertest.IgnoreMetricValues())
+		require.NoError(t, err)
+	})
+	t.Run("Running mongodb 5.0", func(t *testing.T) {
+		t.Parallel()
```
Comment (djaglowski) [inline receiver/mongodbreceiver/integration_test.go:96]:
```suggestion
```

**Review by djaglowski — COMMENTED** [review]
```
go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.
```

File: receiver/mongodbreceiver/integration_test.go (hunk 5 above)
Comment (Frapschen) [reply receiver/mongodbreceiver/integration_test.go:96]: I have changed the second test's port,so it can running parallel.

**Review by djaglowski — APPROVED** [review]
Thank you @Frapschen!

**Pull request merged**